src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/services/service-daemon-list/service-daemon-list.component.html

```diff
@@ -57,23 +57,23 @@
 
 <ng-template #listTpl
              let-events="value">
-  <div *ngIf="events.length == 0 || events == undefined">
+  <div *ngIf="events?.length === 0">
     <span>No data available</span>
   </div>
-  <div *ngIf="events.length != 0 && events != undefined"
-       class="ul-margin">
-    <ul *ngFor="let event of events; trackBy:trackByFn">
-      <li><b>{{ event.created | relativeDate }} - </b>
+  <ul class="list-group list-group-flush" *ngIf="events?.length !== 0">
+    <li class="list-group-item"
+        *ngFor="let event of events; trackBy:trackByFn">
+      <b>{{ event.created | relativeDate }} - </b>
       <span class="badge badge-info">{{ event.subject }}</span><br>
-      <span *ngIf="event.level == 'INFO'">
+      <span *ngIf="event.level === 'INFO'">
       <i [ngClass]="[icons.infoCircle]"
          aria-hidden="true"></i>
       </span>
-      <span *ngIf="event.level == 'ERROR'">
+      <span *ngIf="event.level === 'ERROR'">
       <i [ngClass]="[icons.warning]"
          aria-hidden="true"></i>
       </span>
-      {{ event.message }}</li>
-    </ul>
-  </div>
+      {{ event.message }}
+    </li>
+  </ul>
```

Comment:
Since this kind of blocks are mutually exclusive (if the first block is displayed, the 2nd will be hidden), hence the clause to use here would be an if-else. Also, my personal taste here is that in if-else clauses the first clause should be the desired one:
```html
<ul *ngIf="events?.length else noEventsAvailable" class="list-group list-group-flush">
...
</ul>

<ng-template #noEventsAvailable>
  <div *ngIf="events?.length === 0">
    No data available
  </div>
</ng-template>
```

Reply:
Yes. Updated it.